### ISSUE TITLE
IBM comment 2 on 2.5.1 Pointer Gestures

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag21#456]
IBM is concerned that this wording has implications for non-web software and platforms, and that in its current form it could disrupt the adoption of 2.1 requirements by standards such as Section 508 and EN 301 549.

**Comment by detlevhfischer** [comment]
The addition of the clarification to the Understanding document that the success crtierion appies to author-defined pointer gestures only, which I added in response to [your comment #455](https://github.com/w3c/wcag21/issues/455) may alleviate that concern. If there are other aspects that you fear may have implications for non-web software and platforms, could you be more specific?

**Comment by DavidMacDonald** [comment]
The SC text is this:

> All functionality can be operated with a single untimed pointer gesture unless a multipoint or timed gesture is essential.

I think the language of this SC oversteps its original intention. My recollection is that we wanted to ensure all functionality could be operated *Without* multi point or timed gestures. However this SC language is requiring that all functionality be usable with a mouse (pointer) .

Early on I floated an SC to require everything be usable with a mouse (pointer) but it didn't go very far, which was ok with me.  
http://davidmacd.com/blog/should-WCAG-require-all-functionality-by-mouse.html

Our consensus, as I understand it, is that WCAG 2.1 won't be requiring all functions to be operable with a mouse (pointer), but here it is.

My proposal to address this IBM comment is to roll the requirement of the SC back to something like this:

> All functionality which requires multipoint or timed gestures for operation, can be operated with a single untimed pointer gesture unless a multipoint or timed gesture is essential.

**Comment by detlevhfischer** [comment]
I personally think there is nothing wrong with requiring that all functionality is available for pointer input and that was certainly the original intention of the MATF when devising guideline 2 2.5 Pointer Accessible and this SC. Content is usually pointer accessible  anyway. The edge cases might be things that are controlled via keyboard shortcuts (such as closing a popup only via ESC with no close control). The only concrete problem case that has been put forward by James in issue #495 concern not general pointer accessibility but timed interaction (long press then move to select a menu option). Are there others? Please comment and describe them. I guess we need a WG discussion on that.

**Comment by patrickhlauke** [comment]
> The edge cases might be things that are controlled via keyboard shortcuts (such as closing a popup only via ESC with no close control)

yes, this would pass current 2.1.1 but fail for mouse/touchscreen etc users. The argument usually has been "but authors wouldn't do this as they focus on mouse" / "this affects all users, not just PWD". still, for symmetry and completeness, this should be kept in my opinion.

the other problem that was raised was "do authors now need to provide their own on-screen keyboard when they have things like text entry fields", where really the answer should be no and that they can rely on the OS/UA doing that.

e.g. on mobile devices, where a keyboard is shown under specific conditions - user is focused on a text entry or similar, and on desktop you generally have a keyboard app - like on windows 10, where you can summon the on-screen keyboard and even have that control directly in your system tray (see image)

> ![capture](https://user-images.githubusercontent.com/895831/31674807-adc96306-b35a-11e7-87cf-c9d48d1588db.PNG)

**Comment by detlevhfischer** [comment]
Agree with @patrickhlauke - wrote in the same vein here, in answering @GreggVan's comment in issue #356

**Comment by joshueoconnor** [comment]
@mbgower Could you reply to @detlevhfischer with some more specific examples? Thanks.

**Comment by mbgower** [comment]
What happens with a voice-activated system? Does this wording now cause the author to not only provide a keyboard equivalent, but also a mouse equivalent? Do we really want to force the ability to maintain a pointer controller for all UIs?

**Comment by GreggVan** [comment]
This one still confuses me.

Please look at the following an let me know where I have it wrong.





the purpose of 2.1.1 is to make sure that people who can only use alternate keyboards ( who do not have the ability to use pointers/mice) are able to use the computer.

What is the logic for 2.5.1?    To provide access for people who can only use pointers/mice but can’t use a keyboard?      
Would not an on screen keyboard provide this functionality? 

For mobile devices - where you can swap in new keyboards (true of both Android and iPhone) — you could swap in a full function on screen keyboard  instead of the one usually provided.  you would then have full access to the content using only the “keyboard access” provision.     So if the author met 2.1.1 — they would automatically meet  2.5.1 by relying on the on screen keyboard.    So no point in 2.5.1

If you say — the author can’t rely on the on screen keyboard —   then that would mean the author would need to provide their own on screen keyboard on every screen.


So either the provision is duplicative of 2.1.1  (i.e. meeting 2.1.1 automatically satisfies 2.5.1  — using a full on screen keyboard on the device 

OR

you say that the author can’t rely on the on screen keyboard and the author must provide one themselves for each page.  No?


[you can’t say that doing it by pointer alone means keyboard only is not needed because there are many more that can only use a keyboard or alternate keyboard  than there are people who can ONLY use a pointer/mouse. 



What am I missing?


Gregg

**Comment by awkawk** [comment]
We made the change suggested by David in #456 to clarify this concern.
